packages/pages/src/components/image/image.tsx

```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import { useEffect, useRef, useState } from "react";
+import { ImageProps, ImageLayout } from "./types";
+
+const MKTGCDN_URL_REGEX = /(https?:\/\/a.mktgcdn.com\/p\/)(?<uuid>.+)\/(.*)/;
+
+/**
+ * Renders an image based from the Yext Knowledge Graph. Example of using the component to render
+ * simple and complex image fields from Yext Knowledge Graph:
+ * ```
+ * import { Image } from "@yext/pages/components";
+ *
+ * const simpleImage = (<Image image={document.logo} />);
+ * const complexImage = (<Image image={document.photoGallery[0]} />);
+ * ```
+ *
+ * @public
+ */
+export const Image = ({
+  image,
+  className,
+  width,
+  height,
+  aspectRatio,
+  layout = ImageLayout.INTRINSIC,
+  placeholder,
+  imgOverrides,
+  style = {},
+}: ImageProps) => {
+  const imgRef = useRef<HTMLImageElement>(null);
+  const [imgLoaded, setImgLoaded] = useState(false);
+
+  useEffect(() => {
```

Comment:
```js
const useIsImageLoaded = (): [React.Ref<HTMLImageElement>, boolean] => {
    const imgRef = useRef<HTMLImageElement>(null);
    const [imgLoaded, setImgLoaded] = useState(false);
    
    useEffect(() => {
       if (imgRef.current?.complete) {
      setImgLoaded(true);
    }
  }, []);

  return imgRef, isImageLoaded;
}
```

Reply:
then once this is a custom hook move it outside the view;

Reply:
nvm we can't do this as nicely since we have to call the setter in the img component itself